### ISSUE TITLE
Add compression for  curve snapshot

### ISSUE THREAD

**Issue by xu-chaojie** [opencurve/curve#2223]
**Is your feature request related to a problem? (你需要的功能是否与某个问题有关?)**

**Describe the solution you'd like (描述你期望的解决方法)**

**Describe alternatives you've considered (描述你想到的折衷方案)**

**Additional context/screenshots (更多上下文/截图)**

**Comment by xu-chaojie** [comment]
Description:  
Curve snapshots are currently uploaded to s3, and no compression is used. If you add compression to curve snapshots, there will be great benefits。
Expected outcomes:
1） Detailed Design Documentation
2） Realize curve snapshot compression upload and download decompression
3） Add unit tests and integration test cases
4） Merge your PR into the opencurve repo

Recommended Skills: C++
Mentor(s):  xu-chaojie
Expected project size:
Difficulty: Medium

**Comment by MizukiCry** [comment]
Hi @xu-chaojie. I'm interested in contibruting to this project. As a newbie to open source, Do I need experience with curve? And are there some materials I should learn to have a better understanding of this project?

**Comment by Cyber-SiKu** [comment]
@MizukiCry

**Comment by Cyber-SiKu** [comment]
@MizukiCry 
This is the topic we prepared for [google summer](https://summerofcode.withgoogle.com/), if you are interested, you can sign up from there. This is [timetable](https://developers.google.com/open-source/gsoc/timeline?hl=zh-cn), registration has not started yet, if you are interested, you can start now, write plan or something.

**Comment by caoxianfei1** [comment]
@MizukiCry Do you want to keep doing this？Let us know your plan and work that communicate together.

**Comment by MizukiCry** [comment]
> 

Yes, I'd love to of course. Recently I'm reading the source code of the snapshot part and learning some of the compression algorithms. I'll write the proposal after I understand it.

**Comment by MizukiCry** [comment]
> @MizukiCry Do you want to keep doing this？Let us know your plan and work that communicate together.

Yes, I'd love to of course. Recently I'm reading the source code of the snapshot part and learning some of the compression algorithms. I'll write the proposal after I understand it.

**Comment by MizukiCry** [comment]
@xu-chaojie
Hi. After studying for some time, I have some questions about it.


- I've heard that the compression module is independent to the whole curve project. So do I need to read the source code of curve for a better understanding about the task? I've tried to read the code before but I'm a little confusing about this large project. If I have to, could you give me some hints that how could I get started with it?

- In order to realize the compression feature, I have to know different algorithms. By the time I start to realize this feature, should I complete all the algorithm by myself or I can use some open source code?


Besides, is there a way to get in touch with you? I tried to send an e-mail but seems failed.

Please let me know what I should do if I misunderstood the target.

**Comment by Kaushik-Nishant** [comment]
Hii, I want to contribute to this project. Is there space for me? I am also a newbie. Please guide me on how to get started.

**Comment by ShivangShandilya** [comment]
Greetings everyone, I am quite intrigued by this project. Can anyone help me to get started with this project like what should I know about curves?

**Comment by xu-chaojie** [comment]
You can participate in this issue through google summer, our documents and related ppt are at https://github.com/opencurve/curve-meetup-slides, you can also see the source code analysis document at: https://github.com/opencurve /curve/wiki/Curve%E6%BA%90%E7%A0%81%E5%8F%8A%E6%A0%B8%E5%BF%83%E6%B5%81%E7%A8%8B%E6 %B7%B1%E5%BA%A6%E8%A7%A3%E8%AF%BB

**Comment by caoxianfei1** [comment]
> You can participate in this issue through google summer, our documents and related ppt are at https://github.com/opencurve/curve-meetup-slides, you can also see the source code analysis document at: https://github.com/opencurve/curve/wiki/Curve%E6%BA%90%E7%A0%81%E5%8F%8A%E6%A0%B8%E5%BF%83%E6%B5%81%E7%A8%8B%E6%B7%B1%E5%BA%A6%E8%A7%A3%E8%AF%BB

@ShivangShandilya @Kaushik-Nishant you can follow above document.

**Comment by caoxianfei1** [comment]
> You can participate in this issue through google summer, our documents and related ppt are at https://github.com/opencurve/curve-meetup-slides, you can also see the source code analysis document at: https://github.com/opencurve/curve/wiki/Curve%E6%BA%90%E7%A0%81%E5%8F%8A%E6%A0%B8%E5%BF%83%E6%B5%81%E7%A8%8B%E6%B7%B1%E5%BA%A6%E8%A7%A3%E8%AF%BB

@ShivangShandilya @Kaushik-Nishant you can follow above document and glad to question here if you need help.

**Comment by aavhad1910** [comment]
@xu-chaojie Is this still open? I am planning to submit a proposal here.

**Comment by ilixiaocui** [comment]
> @xu-chaojie Is this still open? I am planning to submit a proposal here.

It is still in progress.

**Comment by aavhad1910** [comment]
> > @xu-chaojie Is this still open? I am planning to submit a proposal here.
> 
> It is still in progress.

Hi @xu-chaojie , I am not able to submit the proposal now using the gsoc portal. How do i proceed to submit one here?

**Comment by wuhongsong** [comment]
> > > @xu-chaojie Is this still open? I am planning to submit a proposal here.
> > 
> > 
> > It is still in progress.
> 
> Hi @xu-chaojie , I am not able to submit the proposal now using the gsoc portal. How do i proceed to submit one here?

sorry miss it, @xu-chaojie  have a look
